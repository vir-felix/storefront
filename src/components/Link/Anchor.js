/* Anchor gets its content from the `children` prop */
/* eslint-disable jsx-a11y/anchor-has-content */
import React, { Component } from "react";
import PropTypes from "prop-types";
import track from "lib/tracking/track";

@track((ownProps) => ({
  component: "Anchor",
  url: ownProps.href
}))
export default class Anchor extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onAnchorClick: PropTypes.func,
    onClick: PropTypes.func
  }

  static defaultProps = {
    onAnchorClick: () => {},
    onClick: () => {}
  }

  @track(() => ({
    action: "Link Clicked"
  }))
  handleClick = (event) => {
    event.preventDefault();
    this.props.onClick(event);
    this.props.onAnchorClick(event);
  }

  @track(() => ({
    action: "Link Enter Key Down"
  }))
  handleKeyDown = (event) => {
    event.preventDefault();
    if (event.key === "Enter") {
      this.props.onClick(event);
      this.props.onAnchorClick(event);
    }
  }

  render() {
    const {
      onAnchorClick, // eslint-disable-line
      ...props
    } = this.props;

    return (
      <a
        {...props}
        onClick={this.handleClick}
        onKeyDown={this.handleKeyDown}
        role="link"
        tabIndex={0}
      />
    );
  }
}
