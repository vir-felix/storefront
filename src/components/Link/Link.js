/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link as NextLink } from "routes";
import { withStyles } from "material-ui/styles";
import classNames from "classnames";

const styles = ({
  link: {
    textDecoration: "none"
  }
});

@withStyles(styles)
class Link extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    classes: PropTypes.object.isRequired
  }

  render() {
    const { classes, children, className, ...props } = this.props;

    return (
      <NextLink {...props}>
        <a className={classNames(classes.link, className)}>
          {children}
        </a>
      </NextLink>

    );
  }
}

export default Link;
