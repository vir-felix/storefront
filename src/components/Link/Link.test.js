import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Link from "./";

test("basic snapshot", () => {
  const component = renderer.create(<Link route="/next">Linked</Link>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
