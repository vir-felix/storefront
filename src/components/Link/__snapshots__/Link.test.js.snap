// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<a
  className=""
  href="/next"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="link"
  tabIndex={0}
>
  Linked
</a>
`;
